runtime/s390x.S

```diff
@@ -800,7 +795,7 @@ LBL(caml_resume_1):
         SWITCH_OCAML_STACKS(%r9, %r2)
         lgr     %r2, %r4
         br      %r5
```

Comment:
```suggestion
        br      %r6
```
Since the code pointer is now loaded in `r6`. I'll try to test this branch on an s390x system ASAP and see if there are any other fixes needed beyond this.

Reply:
All tests pass on s390x with this fix.